Jenkinsfile

```diff
@@ -48,129 +44,103 @@ pipeline {
+                    }
+                }
+                stage ('Windows') {
+                    agent { label 'server-2016-large'}
+                    steps {
+                        timeout(time: 3, unit: 'HOURS') {
+                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}') {
+                                bat 'mvn clean install -B -T 1C -pl !%ITESTS%'
+                                bat 'mvn install -B -pl %ITESTS% -nsu'
                             }
                         }
                     }
-                )
+                }
             }
         }
         stage('Security Analysis') {
-            steps {
-                parallel(
-                    owasp: {
-                        node('linux-large') {
-                            retry(3) {
-                                checkout scm
-                            }
-                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
-                                script {
-                                    // If this build is not a pull request, run full owasp scan. Otherwise run incremntal scan
-                                    if (env.CHANGE_ID == null) {
-                                        sh 'mvn install -q -B -Powasp -DskipTests=true -DskipStatic=true -pl !$DOCS'
-                                    } else {
-                                        sh 'mvn install -q -B -Powasp -DskipTests=true -DskipStatic=true -pl !$DOCS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
-                                    }
+            parallel {
+                stage ('Owasp') {
+                    steps {
+                        withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                            script {
+                                // If this build is not a pull request, run full owasp scan. Otherwise run incremntal scan
+                                if (env.CHANGE_ID == null) {
+                                    sh 'mvn install -q -B -Powasp -DskipTests=true -DskipStatic=true -pl !$DOCS'
+                                } else {
+                                    sh 'mvn install -q -B -Powasp -DskipTests=true -DskipStatic=true -pl !$DOCS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
                                 }
                             }
                         }
-                    },
-                    nodeJsSecurity: {
-                        node('linux-small') {
-                            retry(3) {
```

Comment:
@mcalcote are we losing the retry functionality by switching away from the scripted syntax? Is there a way we can advise this still?

Reply:
Adding the retry logic back into the top level agent as well as each stage where we use an alternate agent.